src/java/com/wolfssl/provider/jsse/WolfSSLSocket.java

```diff
@@ -1150,7 +1150,7 @@ synchronized public void startHandshake() throws IOException {
             try {
                 ret = EngineHelper.doHandshake(0, this.getSoTimeout());
             } catch (SocketTimeoutException e) {
-                throw new IOException(e);
+                throw new SocketTimeoutException("in doHandshake");
```

Comment:
It may be better to just throw the original SocketTimeoutException here instead of a new one.  Then, instead just add a debug log that prints out that we got a timeout in doHandshake():

```
try {
    ret = EngineHelper.doHandshake(0, this.getSoTimeout());
} catch (SocketTimeoutException e) {
    WolfSSLDebug.log(getClass(), WolfSSLDebug.INFO,
                     "got socket timeout in doHandshake()");
    throw e;
}
```